make/find-tools.mk

```diff
@@ -13,6 +13,22 @@ ifeq ($(VERBOSE),1)
 $(info Searching for required executables: $(REQUIRED_EXECUTABLES)...)
 endif
 K := $(foreach exec,$(REQUIRED_EXECUTABLES),\
-        $(if $(shell which $(exec) 2>/dev/null),some string,$(error "ERROR: No "$(exec)" binary found in in PATH!")))
+        $(if $(shell which $(exec) 2>/dev/null),some string,$(error "ERROR: No "$(exec)" binary found in PATH!")))
 endif
 endif
+
+# Define the message used for the sed version check below
+define SED_MSG
+
+ERROR: The current version of sed will cause the deploy script to fail. Try using gnu-sed instead.
+
+eg.
+brew install gnu-sed
+export PATH="/usr/local/opt/gnu-sed/libexec/gnubin:$$PATH
+
+endef
+
+# On MacOS, the default sed install will cause a malformed target YAML, use gnu-sed instead
+ifeq ($(shell uname -s),Darwin)
+$(if $(shell sed --version 2&>1 > /dev/null || echo "bad sed"),$(error ${SED_MSG}))
+endif
```

Comment:
yes, or fix our scripts to make them POSIX compatible (I'm too lazy and I don't have MacOS :-))

Reply:
Yeah, I was planning to take a look at that at some point. I can do that in a follow-up PR if I get around to it.